### ISSUE TITLE
Pool page is blank - LSAT paid w/ no notice or confirmation

### ISSUE THREAD

**Issue by sangaman** [lightninglabs/lightning-terminal#134]
I just downloaded the latest LiT release and started running it using an existing "remote" lnd (running on the same machine). It started up fine and I was able to pull up the terminal in my browser. The terminal page looks fine, but when I go to the "Lightning Pool" page (my main motivation for trying this out) I get a blank page that only says "Lightning Pool" at the top. I don't see any errors in the litd logs, I've included the relevant snippet from the first time I visited the pool page.

I also realized after the fact that my node paid a 1000 sat invoice for an LSAT token upon visiting the pool page - which I assume is to retrieve information about the pool bid/ask orders. It's a small but non-trivial amount and it seems like poor UX that it was paid automatically without any confirmation or even notice on my behalf. It's also unfortunate that I don't know exactly what the LSAT token is for, or how long it's good for.

Logs:

```
2020-11-03 00:44:28.386 [INF] LITD: Handling static file request: /static/js/6.84ee280e.chunk.js
2020-11-03 00:44:28.472 [INF] LITD: Handling gRPC web request: /poolrpc.Trader/ListAccounts
2020-11-03 00:44:28.477 [INF] LITD: Handling gRPC web request: /poolrpc.Trader/ListOrders
2020-11-03 00:44:28.590 [ERR] AUCT: [/poolrpc.ChannelAuctioneer/Terms]: rpc error: code = Internal desc = payment required
2020-11-03 00:44:28.590 [INF] LSAT: Payment of LSAT token is required, paying invoice
2020-11-03 00:44:32.059 [INF] LNDC: Payment 8e141a947e35459d3e2406b8d24da42c95ab0752d9d48753ecaa217f053ea922 completed
2020-11-03 00:45:48.430 [INF] LITD: Handling static file request: /pool
2020-11-03 00:45:48.539 [INF] LITD: Handling static file request: /static/js/3.679bad4f.chunk.js
2020-11-03 00:45:48.540 [INF] LITD: Handling static file request: /static/js/main.8c764815.chunk.js
2020-11-03 00:45:48.879 [INF] LITD: Handling gRPC web request: /lnrpc.Lightning/GetInfo
2020-11-03 00:45:48.928 [INF] LITD: Handling static file request: /static/js/6.84ee280e.chunk.js
2020-11-03 00:45:48.999 [INF] LITD: Handling gRPC web request: /poolrpc.Trader/ListAccounts
2020-11-03 00:45:49.000 [INF] LITD: Handling gRPC web request: /poolrpc.Trader/ListOrders
2020-11-03 00:45:49.054 [INF] LITD: Handling gRPC web request: /lnrpc.Lightning/GetInfo
2020-11-03 00:45:49.156 [INF] LITD: Handling gRPC web request: /lnrpc.Lightning/ListChannels
2020-11-03 00:45:49.189 [INF] LITD: Handling gRPC web request: /lnrpc.Lightning/SubscribeTransactions
2020-11-03 00:45:49.237 [INF] LITD: Handling gRPC web request: /lnrpc.Lightning/SubscribeChannelEvents
2020-11-03 00:45:49.286 [INF] LITD: Handling gRPC web request: /looprpc.SwapClient/Monitor
2020-11-03 00:45:49.305 [INF] LITD: Handling gRPC web request: /lnrpc.Lightning/GetChanInfo
2020-11-03 00:45:49.311 [INF] LITD: Handling gRPC web request: /looprpc.SwapClient/ListSwaps
2020-11-03 00:45:49.313 [INF] LOOPD: Monitor request received
2020-11-03 00:45:49.367 [INF] LITD: Handling gRPC web request: /lnrpc.Lightning/ChannelBalance
2020-11-03 00:45:49.479 [INF] LITD: Handling gRPC web request: /lnrpc.Lightning/WalletBalance
2020-11-03 00:47:49.414 [INF] LITD: Handling static file request: /static/js/7.b59c0e8a.chunk.js
2020-11-03 00:48:05.057 [INF] LITD: Handling gRPC web request: /poolrpc.Trader/ListAccounts
2020-11-03 00:48:05.058 [INF] LITD: Handling gRPC web request: /poolrpc.Trader/ListOrders
```

I can provide system details if that would be helpful.

**Comment by jamaljsr** [comment]
Hey @sangaman our bad on this one. The UI for Pool is not completed yet and should not have been included in the downloadable binaries. We only intended to release the bundled `poold` command line version. We will upload new binaries shortly that does not display this link in the navbar.

Regarding the LSAT token, this is a security measure that we have implemented on the auctioneer server as a DDOS prevention. You can read more about LSATs on https://lsat.tech if you are interested. At today's BTC exchange rate, the 1000 sats are about $0.13 USD. If you'd like, I'd be more than happy to return these funds to you. Just PM me on [twitter](https://twitter.com/jamaljsr) or on the LND slack @jamaljsr with an invoice.

**Comment by sangaman** [comment]
> Hey @sangaman our bad on this one. The UI for Pool is not completed yet and should not have been included in the downloadable binaries. We only intended to release the bundled `poold` command line version. We will upload new binaries shortly that does not display this link in the navbar.
> 
> Regarding the LSAT token, this is a security measure that we have implemented on the auctioneer server as a DDOS prevention. You can read more about LSATs on [lsat.tech](https://lsat.tech) if you are interested. At today's BTC exchange rate, the 1000 sats are about $0.13 USD. If you'd like, I'd be more than happy to return these funds to you. Just PM me on [twitter](https://twitter.com/jamaljsr) or on the LND slack @jamaljsr with an invoice.

Thanks for the quick response. I certainly understand the purpose/need for the LSAT token, my concern is mainly that the LSAT is paid without any sort of notice/confirmation. I figure ideally there'd be a prompt, like "pay X sats to retrieve pool orders?" or something. Or at a minimum some sort of notice about what was paid and what/how long it is good for. It also seems like it could even be a security vulnerability if the software is automatically paying whatever invoice it receives, e.g. could a rogue auctioneer set the LSAT price to 1M sats and have them paid automatically as well?

I'm not aware if the same behavior is present in the command line, but I'd hope that there'd be some sort of confirmation/user interaction there as well before paying, similar to the `lncli` confirmation anytime a user pays an invoice. Regardless I'll try it out soon, thanks again for the help.

**Comment by guggero** [comment]
Oops, my bad. The release binaries have been fixed now and the Pool page isn't included anymore.

Going to change the title of the issue to track the feature request of adding an LSAT payment confirmation.

**Comment by jamaljsr** [comment]
> Thanks for the quick response. I certainly understand the purpose/need for the LSAT token, my concern is mainly that the LSAT is paid without any sort of notice/confirmation. I figure ideally there'd be a prompt, like "pay X sats to retrieve pool orders?" or something. Or at a minimum some sort of notice about what was paid and what/how long it is good for. It also seems like it could even be a security vulnerability if the software is automatically paying whatever invoice it receives, e.g. could a rogue auctioneer set the LSAT price to 1M sats and have them paid automatically as well?

This is a valid point. I agree we should prompt for permission to make this LSAT payment. We'll be sure to add this into the UI. Thanks for the feedback 👍

**Comment by DarthMobius** [comment]
I also experienced the same 'bug' as sangaman. My pub node id is 033aa5bf2296e494f76c3ce2765cc5a9802f4933362017ba332c140575a7c6a1e8 and after I installed litd and opened the pool app, I noticed an unauthorized withdrawal. 

It's interesting that now this already quadrupled in USD terms from Nov 3 when this issue was first opened and who knows what it will be a year from now.  Not too crazy about the design decision to deposit an arbitrary amount to a predefined node (this is just waiting to be exploited in a bad way to drain user funds) - but mostly by lack of any type of documentation about how and why this is happening. (I get the why after reading up on LSAT, but it's quite unnerving to see funds withdrawn without any kind of user prompt, etc.).

If you see more adoption of pool, I guarantee this will become a very sticky point and will create lots of frustration in the community. I strongly recommend you fix this ASAP - 1) by presenting a user prompt 2) by setting a min amount at 250 sat .

P.S. Wouldn't mind getting a refund for this either.

**Comment by nocode23** [comment]
Hello,
I upgraded umbrel to version 0.3.4. I opened LT a paid immediately 1000 sat without any confirmation. How is this possible?
Thank you

**Comment by lukechilds** [comment]
Just putting my 2c here, we had quite a few Umbrel users asking about mysterious payments taken from their Lightning balance. They didn't understand what it was for and after we explained it was an LSAT payment they were surprised they weren't asked for confirmation.

I think just a simple modal explaining why the payment is needed and asking for confirmation that the user is ok with that before taking the balance would be a great addition and prevent confusion.

Even when the amounts are small, if users see payments going out that they don't think are from them, it can be quite alarming.

**Comment by jamaljsr** [comment]
Agreed, I've already begun working on adding this confirmation prompt to the UI. I'll have a PR up by the end of this week. Thanks for additional info.

**Comment by jamaljsr** [comment]
Instead of adding the confirmation prompt to the UI, we've decided to remove the requirement for an LSAT on the server for the RPC calls made on page load. You can now view the Pool page without incurring the 1000sat payment. This should resolve this issue going forward for all new users. Thanks for the feedback everyone.
